src/componentProvider.js

```diff
@@ -31,10 +33,18 @@ export default function componentProvider(config) {
    * Collects component elements and passes them to each instance of the component.
    */
   const init = () => {
-    const componentEls = document.querySelectorAll(selector);
+    let componentEls = [];
```

Comment:
not a big deal, but we have some type inconsistency here, since `document.querySelectorAll` doesn't return an array. I'd advise making this simply:

```suggestion
    let componentEls;
```

Reply:
Ha! This was a last-second change that I waffled about, so I appreciate your thoughts. My eyes were on the `length` check, but if `querySelectorAll` fails we won't reach the length check. Thanks for brining it up!